src/scripts/main.js

```diff
@@ -1,5 +1,34 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+const wallWidth = wall.offsetWidth;
+const wallHeight = wall.offsetHeight;
+const spiderWidth = spider.offsetWidth;
+const spiderHeight = spider.offsetHeight;
+
+document.addEventListener('click', event => {
+  if (!wall.contains(event.target)) {
+    return;
+  }
+
+  spiderWay(event.offsetX, event.offsetY);
 });
+
+function spiderWay(x, y) {
+  if (x > (wallWidth - spiderWidth)) {
+    spider.style.left = `${wall.clientWidth - spiderWidth}px`;
+  } else if (x < spiderWidth) {
+    spider.style.left = `${0}px`;
+  } else {
+    spider.style.left = `${x - spiderHeight / 2}px`;
+  }
+
+  if (y > (wallHeight - spiderHeight)) {
+    spider.style.top = `${wall.clientHeight - spiderHeight}px`;
+  } else if (y < spiderHeight) {
+    spider.style.top = `${0}px`;
+  } else {
+    spider.style.top = `${y - spiderHeight / 2}px`;
+  };
+};
```

Comment:
much better! 
let assign style.top and style.left outside of the function, so it works like this

spider.style.top = `${spiderWay(args)}px`

Reply:
also, these 2 if-blocks are very similar, I think we can create a universal function which will calculate one value depending on which arguments we pass inside of it